app/lib/bk/compat/parsers/bitrise/workflows.rb

```diff
@@ -7,12 +7,15 @@ class Bitrise
       private
 
       def parse_workflow(wf_name, wf_config)
+        agents = { stack: wf_config['stack'] } if wf_config.include?('stack')
         # Extend for other types of steps - approvals/waits for example.
-        BK::Compat::CommandStep.new(label: wf_name, key: wf_name).tap do |cmd_step|
+        BK::Compat::CommandStep.new(label: wf_name, key: wf_name, agents: agents || {}).tap do |cmd_step|
           wf_config['steps'].each do |step|
             step_key, step_config = step.first
-            step_inputs = step_config['inputs'].reduce({}, :merge)
-            cmd_step << load_workflow_step(step_key, step_inputs)
+            step_inputs = (step_config['inputs'] || {}).reduce({}, :merge)
+            translated_step = load_workflow_step(step_key, step_inputs)
+            translated_step.stack = step_config['stack'] if step_config['stack']
```

Comment:
as mentioned in the other file, you should just add it to the `agents` property:

```suggestion
            translated_step.agents['stack'] = step_config['stack'] if step_config.include?('stack')
```